Setup page after installation (#115)
vmware/precaution: docs/setup.md

<!--
    Copyright 2019 VMware, Inc.
    SPDX-License-Identifier: BSD-2-Clause
-->

> # Setup Precaution as a required check

Reviewer: ```suggestion
# Setup a Branch Protection Rule
```

> By default, when Precaution is installed and has access on your project, Precaution creates a check and doesn't stop a pull request from merging even if the result of the check is fail status: 

Reviewer: ```suggestion
By default, after Precaution is installed, it will not automatically prevent the merging of a pull request even if the check resulted in a failure status. 
```

![check_fails](./setup_images/check_fails.png)

> If you want to make Precaution succeed check mandatory do the following steps:

Reviewer: ```suggestion
In order to allow Precaution to prevent pull requests from merging, do the following:
```

## 1. Go to settings

![settings](./setup_images/settings.png)

## 2. Choose branches

![branches](./setup_images/branches.png)

## 3. Create a new branch protection rule 

![add_rule](./setup_images/add_rule.png)

## 4. Setup the branch protection rule 

> If you want to apply this rule to the master branch you will have to fill and click those option:

Reviewer: ```suggestion
If you want to apply this rule to the master branch you will have to check the following options:
```

![branch_protection_rule.png](./setup_images/branch_protection_rule.png)

> Now, when there is a new pull request the new rule will prohibit it from merging into the master branch until a successfully check.

Reviewer 2: ```suggestion
Congratulations! Branch protection is now enabled and will prevent merging code that fails the status check.
```

> Now when there is a new pull request Precaution will prohibit it from merging into the master branch until the Precaution check passes successfully.

Reviewer: ```suggestion
Now, when there is a new pull request the new rule will prohibit it from merging into the master branch until a successfully check.
```

Author: Yep, I did those changes.

<!--
    Copyright 2019 VMware, Inc.
    SPDX-License-Identifier: BSD-2-Clause
-->

# Setup a Branch Protection Rule

By default, after Precaution is installed, it will not automatically prevent the merging of a pull request even if the check resulted in a failure status. 

![check_fails](./images/check_fails.png)

In order to allow Precaution to prevent pull requests from merging, do the following:

> ## 0. Create a pull request

Author: Yep, you are right. Doesn't make sense. I will fix it.

> There should be a pull request after the installation of Precaution in order to setup Precaution as a branch protection rule.

Author: I will fix that makes sense.
